src/wrapper.rs

```diff
@@ -10,57 +10,39 @@ use crate::{Transaction, Transactional, Busy, Ready, Reset, PinState, Error};
 
 /// Wrapper wraps an Spi and Pin object to support transactions
 #[derive(Debug, Clone, PartialEq)]
-pub struct Wrapper<Spi, SpiError, OutputPin, InputPin, PinError, Delay> {
+pub struct Wrapper<Spi, SpiError, CsPin, BusyPin, ReadyPin, ResetPin, PinError, Delay> {
     /// SPI device
     spi: Spi,
 
     /// Chip select pin (required)
-    cs: OutputPin,
+    cs: CsPin,
 
     /// Delay implementation
     delay: Delay,
 
     /// Busy input pin (optional)
-    busy: Option<InputPin>,
+    busy: BusyPin,
 
     /// Ready input pin (optional)
-    ready: Option<InputPin>,
+    ready: ReadyPin,
 
     /// Reset output pin (optional)
-    reset: Option<OutputPin>,
-
+    reset: ResetPin,
 
     pub(crate) err: Option<Error<SpiError, PinError>>,
 }
 
-impl <Spi, SpiError, Output, Input, PinError, Delay> Wrapper<Spi, SpiError, Output, Input, PinError, Delay>  
+impl <Spi, SpiError, CsPin, BusyPin, ReadyPin, ResetPin, PinError, Delay> Wrapper<Spi, SpiError, CsPin, BusyPin, ReadyPin, ResetPin, PinError, Delay>  
 where
     Spi: Transfer<u8, Error = SpiError> + Write<u8, Error = SpiError>,
-    Output: OutputPin<Error = PinError>,
-    Input: InputPin<Error = PinError>,
+    CsPin: OutputPin<Error = PinError>,
     Delay: DelayMs<u32>,
 {
     /// Create a new wrapper object with the provided SPI and pin
-    pub fn new(spi: Spi, cs: Output, delay: Delay) -> Self {
-        Self{spi, cs, delay, err: None, busy: None, ready: None, reset: None}
-    }
-
-    /// Add a busy input to the wrapper object
-    pub fn with_busy(&mut self, busy: Input) {
-        self.busy = Some(busy);
-    }
-
-    /// Add a ready input to the wrapper object
-    pub fn with_ready(&mut self, ready: Input) {
-        self.ready = Some(ready);
-    }
-
-    /// Add a reset output to the wrapper object
-    pub fn with_reset(&mut self, reset: Output) {
-        self.reset = Some(reset);
+    pub fn new(spi: Spi, cs: CsPin, busy: BusyPin, ready: ReadyPin, reset: ResetPin, delay: Delay) -> Self{
+        Self{spi: spi, cs: cs, delay: delay, busy: busy, ready: ready, reset: reset, err: None}
```

Comment:
because the names match you can use the shorter form here if you like
```suggestion
        Self{spi, cs, delay, busy, ready, reset, err: None}
```